components/payloads/cu_sensor_payloads/src/image.rs

```diff
@@ -50,6 +57,28 @@ impl CuImage {
             unsafe { core::slice::from_raw_parts(data.as_ptr() as *const P::Subpixel, data.len()) };
 
         ImageBuffer::from_raw(width, height, raw_pixels)
-            .expect("Failed to create ImageBuffer with CuImage's pixel data.")
+            .ok_or("Could not create the image:: buffer".into())
+    }
+
+    #[cfg(feature = "kornia")]
+    pub fn as_kornia_image<T: Clone, const C: usize>(&self) -> CuResult<Image<T, C>> {
+        let width = self.format.width as usize;
+        let height = self.format.height as usize;
+        let data = self.buffer_handle.as_slice();
+
+        assert_eq!(
+            width, self.format.stride as usize,
+            "STRIDE must equal WIDTH for Kornia compatibility."
+        );
+
+        let size = width * height * C;
+
+        let raw_pixels: &[T] = unsafe {
+            core::slice::from_raw_parts(data.as_ptr() as *const T, data.len() / size_of::<T>())
+        };
+        let img_size: ImageSize = ImageSize { height, width };
+
+        unsafe { Image::from_raw_parts(img_size, raw_pixels.as_ptr(), size) }
```

Comment:
```suggestion
        unsafe { Image::from_raw_parts([width, height].into(), raw_pixels.as_ptr(), size) }
```
to save you from one feature line above

Reply:
and why not passing directly `data.as_ptr()` ?

Reply:
the buffers are u8 on Copper side and I need to pass data: *const T.

I initially went to the route of trying to have the element specified but even that would block a bunch of formats like NV12 where 12 is 12 bits per pixel Oo.. and biplanar :boom: 
Or bayered formats, or compressed buffers etc .. it is endless :). void * it will be for you for now lol.